app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/RoomColumnViewCache.kt

```diff
@@ -0,0 +1,105 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import android.widget.LinearLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import nerd.tuxmobil.fahrplan.congress.models.RoomData
+import nerd.tuxmobil.fahrplan.congress.schedule.RoomColumnViewCache.RoomColumnView
+
+internal interface IRoomColumnViewCache {
+
+    /**
+     * Look for a [RoomColumnView] for the room with the name [roomName].
+     * If such a view does not exist, create one.
+     */
+    fun getOrCreateRoomColumnView(
+        roomName: String,
+        columnWidth: Int
+    ): RoomColumnViewCache.RoomColumnView
+}
+
+/**
+ * Cache for reusing RecycleView/Adapter inside the [FahrplanFragment] when stuff updates in the columns/Rooms
+ */
+internal class RoomColumnViewCache(private val fragment: FahrplanFragment): IRoomColumnViewCache {
+
+    /**
+     * Cache of RecyclerViews
+     * Key is [nerd.tuxmobil.fahrplan.congress.models.RoomData.roomName]
+     */
+    private val cache: MutableMap<String, RoomColumnView> = mutableMapOf()
+
+    private val layoutCalculator = LayoutCalculator(fragment.getNormalizedBoxHeight())
+
+    override fun getOrCreateRoomColumnView(
+        roomName: String,
+        columnWidth: Int,
+    ): RoomColumnView {
+        return cache.getOrPut(roomName) {
+            createFilledRecyclerView(
+                columnWidth
+            )
+        }
+    }
+
+    /**
+     * In case there is no corresponding RecyclerView in the cache, create one and fill it with data
+     */
+    private fun createFilledRecyclerView(
```

Comment:
The function name seems outdated. Here the view is created - but the data is added in `updateData()`.